mce-amd-smca.c

```diff
@@ -815,6 +824,27 @@ static struct smca_bank_name smca_names[] = {
 	[SMCA_GMI_PHY]			= { "Global Memory Interconnect PHY Unit" },
 };
 
+void smca_smu2_ext_err_desc(void)
+{
+	int i, j;
+	int smu2_bits = 62;
+
+	/*
+	 * MCA_CTL_SMU error stings are defined for b'58:59 and b'62
+	 * in MI300A AMD systems. See AMD PPR MCA::SMU::MCA_CTL_SMU
+	 *
+	 * b'0:11 can be decoded from existing array smca_smu2_mce_desc.
+	 * b'12:57 are Reserved and b'58:62 are appended to the
+	 * smca_smu2_mce_desc.
+	 */
+	for (i = 12, j = 0; i < smu2_bits || j < 5; i++, j++) {
+		for ( ; i < 58; i++)
+			smca_smu2_mce_desc[i] = "Reserved";
+
+		smca_smu2_mce_desc[i] = smca_smu2_ext_mce_desc[j];
+	}
```

Comment:
> for (i = 12, j = 0; i < smu2_bits || j < 5; i++, j++) {
>		for ( ; i < 58; i++)

It sounds a very bad idea to use a loop like that with magic numbers (12, 5, 58), as it makes:
- the table obfuscated, as one needs to read this init function to understand the value;
- error prone, as new patches for the table require checking such init code;
- less maintainable, as one needs to update not only the table but also this logic.

Instead, just let the compiler initialize "Reserved" at compile-time using designated initializers [1] like:

```
#define SMA_SMU2_MCE_SIZE 62

static const char * const smca_smu2_mce_desc[SMA_SMU2_MCE_SIZE] = {
       // Set a default value
	[0 ...SMA_SMU2_MCE_SIZE - 1] = "Reserved",
	// Initialize the fields that aren't reserved below here
```
[1] https://gcc.gnu.org/onlinedocs/gcc-3.3.1/gcc/Designated-Inits.html

Reply:
Please track the changes in PR #167.  This PR #144 can be closed.